sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/intervalExpressions.scala

```diff
@@ -641,6 +649,11 @@ case class DivideYMInterval(
       val javaType = CodeGenerator.javaType(dataType)
       val months = left.genCode(ctx)
       val num = right.genCode(ctx)
+      val checkDivideByZero =
```

Comment:
we can add
```
private def divideByZeroCheckCodegen(value: String): String = right.dataType match {
  case _: DecimalType => "if ($value.isZero()) throw ..."
  case _ => "if ($value == 0) throw ..."
}
```

to avoid duplicate code here.

Reply:
and we can move these util functions to `IntervalDivide`

Reply:
Yes.